pkg/sypgp/sypgp.go

```diff
@@ -376,144 +479,136 @@ func GetPassphrase(message string, retries int) (string, error) {
 	return "", errTooManyRetries
 }
 
-// GenKeyPair generates an OpenPGP key pair and store them in the sypgp home folder
-func GenKeyPair(keyServiceURI string, authToken string) (entity *openpgp.Entity, err error) {
+func genKeyPair(name, comment, email, passphrase string) (*openpgp.Entity, error) {
 	conf := &packet.Config{RSABits: 4096, DefaultHash: crypto.SHA384}
 
-	if err = PathsCheck(); err != nil {
-		return
+	entity, err := openpgp.NewEntity(name, comment, email, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// encrypt private key
+	if err = EncryptKey(entity, passphrase); err != nil {
+		return nil, err
+	}
+
+	// Store key parts in local key caches
+	if err = StorePrivKey(entity); err != nil {
+		return nil, err
+	}
+
+	if err = StorePubKey(entity); err != nil {
+		return nil, err
+	}
+
+	return entity, nil
+}
+
+// GenKeyPair generates an OpenPGP key pair and store them in the sypgp home folder
+func GenKeyPair(keyServiceURI string, authToken string) (*openpgp.Entity, error) {
+	if err := PathsCheck(); err != nil {
+		return nil, err
 	}
 
 	name, err := AskQuestion("Enter your name (e.g., John Doe) : ")
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	email, err := AskQuestion("Enter your email address (e.g., john.doe@example.com) : ")
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	comment, err := AskQuestion("Enter optional comment (e.g., development keys) : ")
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	// get a password
 	passphrase, err := GetPassphrase("Enter a passphrase : ", 3)
 	if err != nil {
-		return
+		return nil, err
 	}
 
-	fmt.Printf("Generating Entity and OpenPGP Key Pair...")
-	entity, err = openpgp.NewEntity(name, comment, email, conf)
-	if err != nil {
-		return
-	}
-	fmt.Printf("done\n")
+	fmt.Printf("Generating Entity and OpenPGP Key Pair... ")
 
-	// encrypt private key
-	if err = EncryptKey(entity, passphrase); err != nil {
-		return
+	entity, err := genKeyPair(name, comment, email, passphrase)
+	if err != nil {
+		return nil, err
 	}
 
-	// Store key parts in local key caches
-	if err = StorePrivKey(entity); err != nil {
-		return
-	}
-	if err = StorePubKey(entity); err != nil {
-		return
-	}
+	fmt.Printf("done\n")
 
 	// Ask to push the new key to the keystore
-	pushKeyQ, err := AskQuestion("Would you like to push it to the keystore? [Y,n] : ")
-	if err != nil {
-		return
-	}
+	ans, err := askYNQuestion("y", "Would you like to push it to the keystore? [Y,n] ")
+	switch {
+	case err != nil:
+		fmt.Printf("Not pushing newly created key to keystore: %s\n", err)
```

Comment:
Should this print to stderr?

```suggestion
		fmt.Fprintf(os.Stderr, "Not pushing newly created key to keystore: %s\n", err)
```